src/button.tsx

```diff
@@ -217,6 +219,62 @@ const useDangerButtonStyles = createUseStyles((theme) => ({
     inverse: dangerButtonStyles(theme),
 }));
 
+const renderButtonContent = ({
+    content,
+    defaultIconSize,
+    renderText,
+}: {
+    content: React.ReactNode;
+    defaultIconSize: number;
+    renderText: (text: React.ReactNode) => React.ReactNode;
+}): React.ReactNode => {
+    const length = React.Children.count(content);
+    const childrenArr = React.Children.toArray(content);
+    const resultChildrenArr: Array<React.ReactNode> = [];
+    let accText: Array<React.ReactNode> = [];
+    const flushAccText = () => {
+        resultChildrenArr.push(
+            <React.Fragment key={resultChildrenArr.length}>{renderText(accText)}</React.Fragment>
+        );
+        accText = [];
+    };
+    childrenArr.forEach((element, idx) => {
+        const isFirstChild = idx === 0;
+        const isLastChild = idx === length - 1;
+
+        const isIconElement =
+            React.isValidElement(element) &&
+            typeof element.type !== 'string' &&
+            element.type.name?.startsWith('Icon'); // all mistica icons start with Icon prefix
+
+        if (isIconElement) {
+            if (accText.length) {
+                flushAccText();
+            }
+            const sizeInPx = element.props.size ?? defaultIconSize;
+            resultChildrenArr.push(
+                <div
+                    key={resultChildrenArr.length}
+                    style={{
+                        display: 'flex',
+                        alignItems: 'center',
+                        marginLeft: isFirstChild ? 0 : ICON_MARGIN_PX,
+                        marginRight: isLastChild ? 0 : ICON_MARGIN_PX,
+                    }}
+                >
+                    {React.cloneElement(element, {size: pxToRem(sizeInPx)})}
+                </div>
```

Comment:
mistica icons accept a style prop. The div wrapper shouldn't be needed

Reply:
As you said, it could be a non-mistica icon too, and the component may not have style prop. The extra div always works